mup_scripts/mutransfer_lr/plot.ipynb

```diff
@@ -0,0 +1,154 @@
+      "text/plain": [
+       "<Figure size 1600x600 with 2 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "from tqdm import tqdm\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib as mpl\n",
+    "from matplotlib import cm\n",
+    "import seaborn as sns\n",
+    "sns.set(style='whitegrid')\n",
+    "\n",
+    "parameterizations = [\n",
+    "    ('sp', r'SP'),\n",
+    "    ('mup', r'$\\mu$P'),\n",
+    "]\n",
+    "seeds = [1,2,3]\n",
+    "widths = [\n",
+    "    256,\n",
+    "    512,\n",
+    "    1024,\n",
+    "    2048,\n",
+    "]\n",
+    "lrs = [\n",
+    "    0.125,\n",
+    "    0.0625,\n",
+    "    0.03125,\n",
+    "    0.015625,\n",
+    "    0.0078125,\n",
+    "    0.00390625,\n",
+    "    0.001953125,\n",
+    "    0.0009765625,\n",
+    "    0.00048828125,\n",
+    "    0.000244140625,\n",
+    "    0.0001220703125,\n",
+    "    0.00006103515625,\n",
+    "    0.00003051757812,\n",
+    "]\n",
+    "class MplColorHelper:\n",
+    "\n",
+    "    def __init__(self, cmap_name, start_val, stop_val):\n",
+    "        self.cmap_name = cmap_name\n",
+    "        self.cmap = plt.get_cmap(cmap_name)\n",
+    "        self.norm = mpl.colors.Normalize(vmin=start_val, vmax=stop_val)\n",
+    "        self.scalarMap = cm.ScalarMappable(norm=self.norm, cmap=self.cmap)\n",
+    "\n",
+    "    def get_rgb(self, val):\n",
+    "        return self.scalarMap.to_rgba(val)\n",
+    "\n",
+    "\n",
+    "color_helper = MplColorHelper('viridis', 0, len(widths)-1)\n",
+    "n_cols = 2#len(parameterizations)\n",
+    "n_rows = 1\n",
+    "fig, axes = plt.subplots(n_rows, n_cols, figsize=(8*n_cols, 6*n_rows))\n",
+    "# axes = axes.reshape((n_rows, n_cols))\n",
+    "\n",
+    "for parameterization_idx, (parameterization, parameterization_str) in enumerate(parameterizations):\n",
+    "    ax = axes[parameterization_idx]\n",
+    "    optimal_lrs = []\n",
+    "    optimal_losses = []\n",
+    "    for width_idx, width in enumerate(widths):\n",
+    "        mean_losses = []\n",
+    "        sem_losses = []\n",
+    "        lrs_to_plot = []\n",
+    "        for lr in lrs:\n",
+    "            losses = []\n",
+    "            for seed in seeds:\n",
+    "                job_name = f'width{width}_depth2_seed{seed}_lr{lr:.20f}'.rstrip('0')\n",
+    "                csv_path = os.path.join(parameterization, 'out', job_name, 'log.csv')\n",
+    "                if os.path.exists(csv_path):\n",
+    "                    ckpt_df = pd.read_csv(csv_path)\n",
+    "                    losses.append(ckpt_df['train/loss'].mean())\n",
```

Comment:
Ideally we plot something like
`losses.append(ckpt_df['train/loss'].min())` or `losses.append(ckpt_df['train/loss'].ewm(alpha=0.9).mean().values[-1])` instead once the OWT test has more steps and is larger scale

Reply:
went with `losses.append(ckpt_df['train/loss'].ewm(alpha=0.9).mean().values[-1])`